### PR TITLE
Add comments from incidents to responders

### PR DESCRIPTION
**Pull request by yashmehrotra**

### CODE REVIEW

File: events/events.go [hunk 1]
```diff
@@ -146,3 +157,57 @@ func reconcileResponderEvent(tx *gorm.DB, event api.Event) error {
...
+			// For now, we are only logging the error and moving on
+			// In future, we would want to associate error messages with responderType
+			// and when reprocessing, handle it only for specific responders
+			logger.Errorf("Error adding comment to responder:%s %v", responder.Properties["responderType"], err)
+			continue
+		}
+
+		// Insert into comment_responders table
+		if externalID != "" {
+			insertQuery := `
```
Comment (moshloop) [inline events/events.go:201]: Why not just inline the query ?

Comment (yashmehrotra) [reply events/events.go:201]: primarily for readability, keeping the query and the statement separate

File: events/events.go [hunk 2]
```diff
@@ -146,3 +157,57 @@ func reconcileResponderEvent(tx *gorm.DB, event api.Event) error {
...
+	var responders []api.Responder
+	commentRespondersQuery := `
+        SELECT * FROM responders WHERE incident_id IN (
+            SELECT incident_id FROM comments WHERE id = ?
+        )
+    `
+	err := tx.Raw(commentRespondersQuery, commentID).Preload("Team").Find(&responders).Error
+	if err != nil {
+		return err
+	}
```
Comment (moshloop) [inline events/events.go:175]:
```suggestion
	if err := tx.Raw(commentRespondersQuery, commentID).Preload("Team").Find(&responders).Error; err != nil {
		return err
	}
```

File: events/events.go [hunk 3]
```diff
@@ -146,3 +157,57 @@ func reconcileResponderEvent(tx *gorm.DB, event api.Event) error {
...
+			externalID, err = responderPkg.NotifyJiraResponderAddComment(responder, commentBody)
+		case responderPkg.MSPlannerResponder:
+			externalID, err = responderPkg.NotifyMSPlannerResponderAddComment(responder, commentBody)
+		default:
+			continue
+		}
+
+		if err != nil {
+			// For now, we are only logging the error and moving on
+			// In future, we would want to associate error messages with responderType
```
Comment (moshloop) [inline events/events.go:193]:
```suggestion
			// TODO associate error messages with responderType
```

File: events/events.go (hunk 1 above)
Comment (moshloop) [reply events/events.go:201]: I think it makes it less readable, if it is a very long query that is one thing, but if the query can fit on single line, and inlined in the parameters its fewer lines to read

File: responder/events.go [hunk 4]
```diff
@@ -96,3 +118,54 @@ func NotifyMSPlannerResponder(responder api.Responder) (string, error) {
...
+	if err != nil {
+		return "", err
+	}
+
+	err = client.AddComment(responder.ExternalID, comment)
+
+	// MS Graph API does not return the ID of the comment created
+	// so we just return a constant string for now
+	return "posted", err
+}
```
Comment (moshloop) [inline responder/events.go:171]: I think all of this should go into the msplanner package,  and then maybe add a registry of responderType to responder interface

File: api/types.go [hunk 5]
```diff
@@ -10,12 +10,18 @@ import (
...
 type Comment struct {
-	Body      string    `json:"body"`
-	CreatedBy string    `json:"created_by"`
-	CreatedAt time.Time `json:"created_at"`
+	Body       string    `json:"body"`
+	CreatedBy  string    `json:"created_by"`
+	CreatedAt  time.Time `json:"created_at"`
+	IncidentID uuid.UUID `json:"incident_id"`
+	Incident   Incident  `json:"incident"`
 }
```
Comment (moshloop) [inline api/types.go:25]: Wdo we have both?

File: events/events.go (hunk 1 above)
Comment (yashmehrotra) [reply events/events.go:201]: Cool

File: events/events.go (hunk 3 above)
Comment (yashmehrotra) [reply events/events.go:193]: 👍🏽

File: events/events.go (hunk 2 above)
Comment (yashmehrotra) [reply events/events.go:175]: 👍🏽

File: responder/events.go (hunk 4 above)
Comment (yashmehrotra) [reply responder/events.go:171]: Yeah, moving it to a package is better, I'll pick up the interface part in the config task

File: api/types.go (hunk 5 above)
Comment (yashmehrotra) [reply api/types.go:25]: yeah, removing `incident_id` for now

File: db/migrations/004_incidents.sql [hunk 6]
```diff
@@ -89,16 +89,24 @@ CREATE TABLE comments (
   id UUID DEFAULT generate_ulid() PRIMARY KEY,
   created_by UUID NOT NULL ,
   comment text NOT NULL,
-  external_id TEXT NULL, -- A unique identifier for the responder in the external system e.g. Jira ticket id
   incident_id UUID NOT NULL,
-  responder_id UUID NULL,
```
Comment (moshloop) [inline db/migrations/004_incidents.sql:94]: I think we still need a responder_id, this will be the source of the comment

Comment (yashmehrotra) [reply db/migrations/004_incidents.sql:94]: Oh, for comments made from responder and added here ? I guess so, in that case we should put back the external_id as well here.

File: api/types.go [hunk 7]
```diff
@@ -10,12 +10,17 @@ import (
 )
 
 type Incident struct {
+	ID     uuid.UUID `json:"id"`
+	Title  string    `json:"title"`
+	TeamID uuid.UUID `json:"team_id"`
+	Team   Team      `json:"team"`
```
Comment (moshloop) [inline api/types.go:16]: Why both ?

Comment (yashmehrotra) [reply api/types.go]: I initially thought there maybe usecases where we may need teamID but not the entire team object but I don't think thats the case for now, have removed it from here and other places as well

**Pull request merged**